walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,77 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+  /*
+   * @return the Collection of TimeRanges containing all the possible time frames to schedule
+   * the request ensuring that the attandees don't have any conflicts with other events.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> requestAttendeeRanges =
+        filterRequestAttendeeTimeRanges(events, request.getAttendees());
+    Collections.sort(requestAttendeeRanges, TimeRange.ORDER_BY_START);
+    requestAttendeeRanges = combineOverlaps(requestAttendeeRanges);
+    return findMeetingRangesWithNoConflict(requestAttendeeRanges, request.getDuration());
+  }
+
+  /*
+   * @return an ArrayList of all the events with at least one attendee from the request.
+   */
+  private ArrayList<TimeRange> filterRequestAttendeeTimeRanges(
```

Comment:
`filterToRequestAttendeeTimeRanges`? Also good to add more Javadoc on the specific inputs, or not put it as a summary fragment:
```
@param events list of all events
@param requestAttendees list of attendees that we want to get events for
...
```
https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment